python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -293,6 +295,70 @@ def _fit(model, X_df_list, y_df_list, r):
     def _predict(model, X, r):
         return model._predict_get_all(X)
 
+    @staticmethod
+    def _print_summary(model):
+        model.print_summary()
+
+    @staticmethod
+    def _tl_model_handles(model, model_bytes):
+        return model._tl_model_handles(model_bytes=model_bytes)
+
+    @staticmethod
+    def _build_fil_model(model, model_handle):
+        print("model handels in dask func : ", model_handle)
+        return model._build_fil_model(model_handle=model_handle)
+
+    @staticmethod
+    def _read_mod_handles(model, mod_handles):
+        return model._read_mod_handles(mod_handles=mod_handles)
+
+    def print_summary(self):
+        """
+        prints the summary of the forest used to train and test the model
+        """
+        c = default_client()
+        futures = list()
+        workers = self.workers
+
+        for n, w in enumerate(workers):
```

Comment:
Can we just do something like:

```
for rf in self.rfs:
  rf.print_summary()
```

and let Dask do the routing for us?

Reply:
I can obtain the model information from `self.rfs` and then loop through to print the summary of the models

Reply:
I tried to read the model from rfs and then use that model to call the print_summary function but it prints an empty tree when I do so.

Reply:
We can not print the summary or detailed information of the trees without using the client as we use a pointer of the random forest information as the input for print. If we read the pointer value without using the client to load the model then, the pointer value will be different from the one actually used. 
This is also a limitation for single gpu rf as we cannot print the forest information right after loading the model.
```
  del orignal_model
  cu_after_pickle_model = pickle.load(pf)
  cu_after_pickle_model.print_summary()
```
 However, for single gpu forest we will be able to print the model after we run a function which updates the `self.forest` or `self.forest64` in cython.